server/utils/EmbeddingEngines/genericOpenAi/index.js

```diff
@@ -0,0 +1,95 @@
+const { toChunks } = require("../../helpers");
+
+class GenericOpenAiEmbedder {
+  constructor() {
+    if (!process.env.EMBEDDING_BASE_PATH)
+      throw new Error(
+        "GenericOpenAI must have a valid base path to use for the api."
+      );
+    const { OpenAI: OpenAIApi } = require("openai");
+    this.basePath = process.env.EMBEDDING_BASE_PATH;
+    this.openai = new OpenAIApi({
+      baseURL: this.basePath,
+      apiKey: process.env.GENERIC_OPEN_AI_EMBEDDING_API_KEY ?? null,
+    });
+    this.model = process.env.EMBEDDING_MODEL_PREF ?? null;
+
+    // Limit of how many strings we can process in a single pass to stay with resource or network limits
+    this.maxConcurrentChunks = 500;
+
+    // https://platform.openai.com/docs/guides/embeddings/embedding-models
+    this.embeddingMaxChunkLength = 8_191;
```

Comment:
Above docs are irrelevant to the snippet so we should remove them.
`// Refer to your specific model and provider you use this class with to determine a valid maxChunkLength`

Also, we allow setting for chunk length for custom models and it should be used here. Otherwise smaller embedder BERT models will overflow

https://github.com/Mintplex-Labs/anything-llm/blob/c8c618137f2221939748c32d71d131628d53d6e2/server/utils/helpers/index.js#L142